Sec 4816 deprovisioning issues

eswidler opened the pull request:
The changes below bring us up-to-date with the [feature-EnhancedMatch](https://github.com/awslabs/ssosync/tree/feature-EnhancedMatch) branch of the ssosync repo.  This has added functionality for adding multiple groups which we had a hacky workaround for before, as well as support for querying more than 50 users at a time for deprovisioning.

I also added one custom commit at the end of this PR.  This resolves errors when a deprovisioning event occurs.  We needed to re-get the list of users in AWS at one point, since it's possible that they would've been deleted at an earlier step.  This was causing 404's when querying the AWS API.

Noting that this is currently failing unit tests for unrelated issues in the upstream branch.  This work is from an in-progress branch upstream, however, I've tested that this will cover our use cases for now and addresses the deprovisioning concerns.  I will open an issue in the upstream repo to address the deprovioning issue I needed to open a change for, and will eventually redeploy this when the owner merges this branch with passing unit tests.

How this was tested:
* Moved our sandbox AWS account out of our AWS org.  I had to do this since IAM Identity center can only be enabled in root accounts in AWS orgs.
* Created a new instance of IAM Identity center and credentials to test with.
* Created a new SCIM Endpoint Access token for this account.
* Updated samconfig.toml with new SCIM credentials from IAM Identity center
* Made my updates and used `sam build` to prepare for deployment
* Deployed to the sandbox account with `sam deploy`.
* In the cloudformation template, added all google groups we currently sync in our management account to the test account.
* Tested adding and removing several users from a test Google Group and then running the AWS SSO Sync script again would cause all users to be added or removed from the group in AWS.

How this will be deployed:
* Since we no longer store the SCIM Endpoint Access token for our management account in Lastpass and this is a hidden value in the cloudformation template, I'll generate a new set.  We'll continue to not save this on Lastpass.
* Will update samconfig.toml with the new credentials for our management account.
* Will deploy to our management account using `sam deploy`.
* Will verify that all expected deprovisioned users will not exist in AWS Identity Center groups.
* Will move sandbox account back into main AWS org.

Commit:
IN-28: Google client err handling improvement

Commit:
IN-28: Add support for CodePipeline

Commit:
Ignore missing user in group members

ChrisPates committed:
Update buildspec.yml

debug 6

ChrisPates committed:
Update buildspec.yml

debug 7

ChrisPates committed:
Update buildspec.yml

debug 8

ChrisPates committed:
Update buildspec.yml

debug 9

ChrisPates committed:
Update buildspec.yml

debug 10

ChrisPates committed:
Update buildspec.yml

debug 11

ChrisPates committed:
switching to GitHub v2

split into multiple buckets.
adopt CodeStarConnection

ChrisPates committed:
pipeline improvements

* moving all cicd templates and resources under cicd path.
* splitting build stage into builggo and then separate SAM build stages for target environments.

ChrisPates committed:
improving pipeline

create codebuild path for buildspec files

ChrisPates committed:
Update pipeline.yaml

fixing errors

ChrisPates committed:
debugging

buildgo.yml : corrected format error
pipeline.yaml : fix policies

ChrisPates committed:
Update buildgo.yml

adding artifacts for packaging stage

ChrisPates committed:
Update buildgo.yml

escape golint until the pipeline is working.

ChrisPates committed:
updates to codebuild

Addressing issues with insufficient artifacts passing between stages.

ChrisPates committed:
Environment Vars

Moving them to code pipeline action definitions.

ChrisPates committed:
Update pipeline.yaml

Improving resource naming

ChrisPates committed:
restructuring cicd

ChrisPates committed:
Update buildspec.yml

Correct artifact recursion

ChrisPates committed:
various

fix typo in bucket policies
add release stages
minimise instances of buildspec.yml

ChrisPates committed:
Streamline packaging stages

Using regex to change app name in template.yaml as part of package/buildspec.yml

ChrisPates committed:
fixes

added .swp to .gitignore
fixed pipeline.yaml errors
fixed package/buildspec.yml error

ChrisPates committed:
working up to publish stage

Corrected various issues.

Limitation of running sed within buildspec mean regex has to be in hardcoded in the .yml

ChrisPates committed:
change buildspec path

BuildGo stage

ChrisPates committed:
Update pipeline.yaml

Adding Permissions to Staging/Release buckets for codepipeline role.

ChrisPates committed:
debug publish stage

ChrisPates committed:
simplifying

ChrisPates committed:
Update production.yaml

single bucket, minimal buildspecs

ChrisPates committed:
'Production' Pipeline

Working 'production' pipeline.

ChrisPates committed:
golang:1.18.1

To reduce execution time, adding action to build custom container image.

ChrisPates committed:
Build Images

updated pipeline:
- group Actions into Stages
- create Dockerfiles
- create buildimage.yml

ChrisPates committed:
Build Image

debugging go.lang image

ChrisPates committed:
fix up the pipeline

fix use of secondary input artifacts.

ChrisPates committed:
Add Deploy to Staging

ChrisPates committed:
Update buildspec.yml

typo

ChrisPates committed:
Staging Deployment

- Add cross account access
- add output to the staging 'packaged.yaml'

ChrisPates committed:
staging-account

Split secrets out of staging-accounts

ChrisPates committed:
Merge branch 'master' into CodePipeline

ChrisPates committed:
Create outputs.yaml

will be appended to the packaged.yaml during the ReleaseCandidate phase.

ChrisPates committed:
ActionType: Cloudformation

debugging, kms key access

ChrisPates committed:
Update stack.yml

ChrisPates committed:
test stack cleanup

Adding test stack cleanup

ChrisPates committed:
switch from sed to diff

Use diff & patch to adjust packaged.yaml
fix permissions issues with pipeline

ChrisPates committed:
patch files

updating errors with .patch files.

ChrisPates committed:
patch files

Moved creation of template variants to PackageApp stage.

ChrisPates committed:
publish

correcting paths to patch files

ChrisPates committed:
Update buildspec.yml

Correcting stage in patch paths.

ChrisPates committed:
Update buildspec.yml

Added --single-target to goreleaser command. Reduce unnecessary storage and reduce build time.

ChrisPates committed:
Fix .goreleaser for CodeBuild

Fixing mixing github variables in goreleaser builds.

ChrisPates committed:
GitHub Variables

CodeBuild only gathers some of the information from the current commit, so this stage gathers the rest.

ChrisPates committed:
GitHub variables

debugging

ChrisPates committed:
Is a Release? Stage

Determine via gitHub metadata whether this is commit is a release.

ChrisPates committed:
gitvars

minor tweaks

ChrisPates committed:
.goreleaser.yaml

re-instate

ChrisPates committed:
creating pipeline variants

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
v0.0.0-Staging

The SAR app versions.

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update production-account.yaml

ChrisPates committed:
publish stage

remove old staging build from SAR, before publishing new

ChrisPates committed:
SemanticVersion

Move Staging SAR Arn from Publish

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
paramstore & Pipeline Variables

ChrisPates committed:
Switch to CodeBuild environment variables

ChrisPates committed:
switch to CodeBuild

Logging for Deploy Action of Cloudformation is limited, so switching to CodeBuild and aws cli v2.

ChrisPates committed:
Debugging deploy

ChrisPates committed:
Automated test

- smoke test
- check whether isRelease before public release of build.

ChrisPates committed:
Update production-account.yaml

Remove manual release approval.

ChrisPates committed:
updating Docker

Setting up dockerfile to extend codebuild ubuntu/standard:5.0 for this pipeline.

ChrisPates committed:
fix upstream references

ChrisPates committed:
removed buildImages

taking the hit of longer build times , for now.

ChrisPates committed:
Merge pull request #99 from awslabs/CodePipeline

Code pipeline

ChrisPates committed:
CodeStar

Switch to Master branch

ChrisPates committed:
CodeStar

Switch to Master branch

ChrisPates committed:
Delete buildimages.yaml

Remove build images pipeline for now take the hit of installation of tooling on the fly.

ChrisPates committed:
Delete buildimages.yaml

Remove build images pipeline for now take the hit of installation of tooling on the fly.

ChrisPates committed:
Update production-account.yaml

correcting case: Master to master

ChrisPates committed:
Update production-account.yaml

correcting case: Master to master

ChrisPates committed:
Update buildspec.yml

Update Variable name.

ChrisPates committed:
Update buildspec.yml

Update Variable name.

ChrisPates committed:
Merge branch 'pr/63' into CodePipeline

ChrisPates committed:
CodePipeline Lambda execution

Test ability to execute the Lambda within a CodePipeline instance.

ChrisPates committed:
CodePipeline Lambda execution

Test ability to execute the Lambda within a CodePipeline instance.

ChrisPates committed:
Debugging Codepipeline Support

ChrisPates committed:
Debugging Codepipeline Support

ChrisPates committed:
back out pr/63

Back out pr/63.

ChrisPates committed:
back out pr/63

Back out pr/63.

ChrisPates committed:
RUNNING_IN_CODEPIPELINE

ChrisPates committed:
RUNNING_IN_CODEPIPELINE

cmbrehm committed:
bug: --include-groups is broken

cmbrehm committed:
needed to invert

cmbrehm committed:
Merge pull request #78 from DNXLabs/fixup/get-missing-user

fix: Ignore missing user in group members

Commit:
Changes from AWS 2022 Summer intern project

Commit:
Changes from AWS 2022 Summer intern project

ChrisPates committed:
Adding Codepipeline support

Add config flag and new handler

ChrisPates committed:
Adding Codepipeline support

Add config flag and new handler

ChrisPates committed:
Update config.go

ChrisPates committed:
Update config.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Merge branch 'master' into CodePipeline

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
correct staging patch

ChrisPates committed:
correct staging patch

ChrisPates committed:
fixing param export

ChrisPates committed:
fixing param export

ChrisPates committed:
adding invokePermission

ChrisPates committed:
adding invokePermission

ChrisPates committed:
Update staging.patch

Added codepipeline permissions

ChrisPates committed:
Update staging.patch

Added codepipeline permissions

ChrisPates committed:
Update staging.patch

ChrisPates committed:
Update staging.patch

ChrisPates committed:
Add EnvVar Step

ChrisPates committed:
Add EnvVar Step

ChrisPates committed:
Update not Replace envVars

ChrisPates committed:
Update not Replace envVars

ChrisPates committed:
codepipeline detection

ChrisPates committed:
codepipeline detection

ChrisPates committed:
Update buildspec.yml

Trying to resolve import timeout issues

ChrisPates committed:
Update buildspec.yml

Trying to resolve import timeout issues

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Update root.go

ChrisPates committed:
Adding Container Test Stage

Adding Test Stage to prove container/cli execution.

ChrisPates committed:
Adding Container Test Stage

Adding Test Stage to prove container/cli execution.

ChrisPates committed:
Add Container/CLI Test

ChrisPates committed:
Add Container/CLI Test

ChrisPates committed:
split pipeline

Due to limitations of cross account execution, have need to split into two codepipeline instances, on in the production account and one in the staging account, building and testing respectively.

ChrisPates committed:
split pipeline

Due to limitations of cross account execution, have need to split into two codepipeline instances, on in the production account and one in the staging account, building and testing respectively.

ChrisPates committed:
cross pipeline int

refining cross pipeline integration

ChrisPates committed:
cross pipeline int

refining cross pipeline integration

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
update paths

ChrisPates committed:
update paths

ChrisPates committed:
update paths

ChrisPates committed:
update paths

ChrisPates committed:
recursive cp of tests

ChrisPates committed:
recursive cp of tests

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
clean up split pipeline

ChrisPates committed:
clean up split pipeline

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
updating developer

ChrisPates committed:
updating developer

ChrisPates committed:
pipeline clean up

ChrisPates committed:
pipeline clean up

ChrisPates committed:
Merge branch 'master' into pr/103

ChrisPates committed:
Clean up

Make changes to meet linter standards.

ChrisPates committed:
Clean up

Make changes to meet linter standards.

ChrisPates committed:
Create .gitignore

ChrisPates committed:
Create .gitignore

ChrisPates committed:
minor pipeline fixes

ChrisPates committed:
minor pipeline fixes

ChrisPates committed:
CICD tweaks

ChrisPates committed:
CICD tweaks

ChrisPates committed:
CICD bugfixes

ChrisPates committed:
CICD bugfixes

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Merge branch 'master' into IAM-IdStore

ChrisPates committed:
Create .goreleaser.yml

Added back in lost config

ChrisPates committed:
Create .goreleaser.yml

Added back in lost config

ChrisPates committed:
patch fiels

Updating patches

ChrisPates committed:
patch fiels

Updating patches

ChrisPates committed:
Additional Parameters

Added the support required for the additional parameteres into the testing stacks and the secrets cloudformation

ChrisPates committed:
Additional Parameters

Added the support required for the additional parameteres into the testing stacks and the secrets cloudformation

ChrisPates committed:
adding additional cli params to smoke test

ChrisPates committed:
adding additional cli params to smoke test

ChrisPates committed:
Expanding Smoke Test

Adding smokeTests for:
* management account
* delegated admin account
* non-delegated account

Improving error handling under lambda excution.

ChrisPates committed:
Expanding Smoke Test

Adding smokeTests for:
* management account
* delegated admin account
* non-delegated account

Improving error handling under lambda excution.

ChrisPates committed:
CICD improvements

Adding account type selection to testing.yaml to allow for different tests and configurations across: Management/delegated/nondelegated accounts.

ChrisPates committed:
CICD improvements

Adding account type selection to testing.yaml to allow for different tests and configurations across: Management/delegated/nondelegated accounts.

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Expect failure

for NonDelegated account.

ChrisPates committed:
Expect failure

for NonDelegated account.

ChrisPates committed:
Updating README

Adding v2.0.0 release guidance.

ChrisPates committed:
Updating README

Adding v2.0.0 release guidance.

ChrisPates committed:
Update README.md

ChrisPates committed:
Update README.md

ChrisPates committed:
fix release stage

correcting semantic version passing.

ChrisPates committed:
fix release stage

correcting semantic version passing.

ChrisPates committed:
GitTag -> AppVersion

Tweak to release stage to stripe leadin 'v' fromGit.Tag to form semantic version

ChrisPates committed:
GitTag -> AppVersion

Tweak to release stage to stripe leadin 'v' fromGit.Tag to form semantic version

fredericbarthelet committed:
Ignore missing user in group members

fredericbarthelet committed:
Ignore missing user in group members

jferris committed:
Add identityStore:DeleteGroup IAM permission

This action is performed during the sync:
https://github.com/awslabs/ssosync/blob/955e8e23a42340d323e25659c8c03130b5609c80/internal/sync.go#L511

But the template does not give the IAM role permission to delete groups:
https://github.com/awslabs/ssosync/blob/b4352b919996c0dfd4fad03c0e7a8000e5fd5f88/template.yaml#L164

Resolves #116.

jferris committed:
Add identityStore:DeleteGroup IAM permission

This action is performed during the sync:
https://github.com/awslabs/ssosync/blob/955e8e23a42340d323e25659c8c03130b5609c80/internal/sync.go#L511

But the template does not give the IAM role permission to delete groups:
https://github.com/awslabs/ssosync/blob/b4352b919996c0dfd4fad03c0e7a8000e5fd5f88/template.yaml#L164

Resolves #116.

ChrisPates committed:
Update developer.yaml

Make parameter for git branch pass through to GitVar stage.

ChrisPates committed:
Update developer.yaml

Make parameter for git branch pass through to GitVar stage.

ChrisPates committed:
Update buildspec.yml

Add echo of AppVersion to aid debugging.

ChrisPates committed:
Update buildspec.yml

Add echo of AppVersion to aid debugging.

ChrisPates committed:
Update sync.go

Where IAM Identity Center user doesn't have email.Praimary set continue.

ChrisPates committed:
Update sync.go

Where IAM Identity Center user doesn't have email.Praimary set continue.

ChrisPates committed:
Merge branch 'pr/112' into 118-panic-observed-when-trying-to-get-existing-aws-users

ChrisPates committed:
Update sync.go

Correcting comment char.

ChrisPates committed:
Update sync.go

Correcting comment char.

ChrisPates committed:
Update CHANGELOG.md

ChrisPates committed:
Update CHANGELOG.md

ChrisPates committed:
Update sync.go

ChrisPates committed:
enable debug logging in developer pipeline

ChrisPates committed:
Update buildspec.yml

ChrisPates committed:
Update client.go

Add support for comma separate list of group query conditions to iterate through.

ChrisPates committed:
Functional Tests for Groups

Adding Functional Tests Groups:
empty query
single query
multi query
external users (as member of a group)
nested groups

ChrisPates committed:
Update client.go

ChrisPates committed:
Update client.go

ChrisPates committed:
SSOSync-CodeBuild-Test

Add "identitystore:DeleteGroup" permission to test role.

ChrisPates committed:
Correct testcase query string

ChrisPates committed:
Add userMatch implementation

Run userMatch queries, so we can sync users who are not a group member based on the query string in userMatch parameter.

Include support for multiple queries based on a comma separated set of queries in userMatch.

ChrisPates committed:
update README

ChrisPates committed:
Update sync.go

ChrisPates committed:
All '*' in -g & -u

To specify all groups (their members) and all users.

ChrisPates committed:
Setting default to no match

To enable full directory sync but ensure backwards compatible behavior setting defaults to sync nothing for both users and groups.

ChrisPates committed:
Update SmokeTest

Setting SmokeTest to sync all users.

ChrisPates committed:
Update Group Functional Tests

ChrisPates committed:
Update sync_test.go

ChrisPates committed:
Service Name

Update service name.

ChrisPates committed:
Add clarification on IAM credentials for ssosync

"Ensure that the default credentials are for the AWS account you delegated admin for AWS IAM Identity Center to. This should be credntials for permissions limited IAM User, it can't be a user in AWS IAM Identity Center since the sync process is going to make changes. The cloudformation template (template.yaml) in this repository includes an IAM policy that grants the minimal set of rights for SSOSync to function.."

ChrisPates committed:
Update README.md

Adding additional steps for preparing the Google Directory.

ChrisPates committed:
Merge branch 'filter-logic-fix'

ChrisPates committed:
Merge branch 'master' into feature-EnhancedMatch

ChrisPates committed:
Revert "Merge branch 'filter-logic-fix'"

This reverts commit 14265335c8ea22a873b9373965a7380598304609, reversing
changes made to 49d87f760094813b5cc6318ddaa152ac555fde65.

ChrisPates committed:
Merge branch 'master' into pr/77

ChrisPates committed:
Merge branch 'fix-user-exists'

ChrisPates committed:
Merge branch 'master' into feature-EnhancedMatch

ChrisPates committed:
Expand Functional Testing fro groups

ChrisPates committed:
Overhaul of logic to retrieve Google Directory objects

MultiQuery support for GroupMatch/UserMatch, including "*" whole directory
Improved IgnoreGroups/ignoreUsers
implement includeGroups/includeUsers
Added flattening of nested groups
improved handling of external group memberships

eswidler committed:
SEC-4816: Resolve Deprovisioning errors

eswidler commented on internal/sync.go:
```diff
@@ -643,8 +646,9 @@ func (s *syncGSuite) getGoogleDirectory() ([]*admin.Group, []*admin.User, map[st
 
 			// Retrieve the user and add to list of users to sync
-                	uniqueUsers[m.Email] = s.getGoogleUser(m.Email)
-			uniqueMembers[m.Email] = s.getGoogleUser(m.Email)
```
This two-time lookup of the user was causing a huge timesink.  Caching the user saved us over 2 minutes of time for each run of the sync.

eswidler commented on internal/sync.go:
```diff
@@ -409,6 +409,9 @@ func (s *syncGSuite) SyncGroupsUsers() error {
 		}
 	}
 
+	log.Info("Re-get content of AWS Identity Store.  Users may have been deleted.")
+	_, _, awsGroupMembership := s.getIdentityStore()
+
```
We reget the contents of `awsGroupMembership`, since some users may have been removed from groups at an earlier step.  This was causing 404's when looking up the user in AWS when deprovisioning.

yugoPanorama commented:
Custom commit looks 👍 to me.

> How this was tested:
>
> Moved our sandbox AWS account out of our AWS org. I had to do this since IAM Identity center can only be enabled in root accounts in AWS orgs.
> Created a new instance of IAM Identity center and credentials to test with.
> Created a new SCIM Endpoint Access token for this account.
> Updated samconfig.toml with new SCIM credentials from IAM Identity center

What's `samconfig.toml`?

> Made my updates and used sam build to prepare for deployment
> Deployed to the sandbox account with sam deploy.
> In the cloudformation template, added all google groups we currently sync in our management account to the test account.
> Tested adding and removing several users from a test Google Group and then running the AWS SSO Sync script again would cause all users to be added or removed from the group in AWS.

Did this cover the case when there's more than 50 users and one of the last ones gets deleted?

> How this will be deployed:
> 
> Since we no longer store the SCIM Endpoint Access token for our management account, I'll generate a new set. We'll continue to not save this on Lastpass.

This can be answered in a private channel, but I'm curious how this gets managed.

> Will update samconfig.toml with the new credentials for our management account.
> Will deploy to our management account using sam deploy.
> Will verify that all expected deprovisioned users will not exist in AWS Identity Center groups.
> Will move sandbox account back into main AWS org.

eswidler commented:
Addressing your points:

We use the [SAM CLI](https://github.com/aws/aws-sam-cli) to deploy the infrastructure changes here.  Deploying with SAM uses a `samconfig.toml` file which we don't commit to source code, since it has sensitive creds.  [On panopedia](https://panoramaed.atlassian.net/wiki/spaces/ENG/pages/2833842186/AWS+SSO+Sync#Deploy-to-AWS) in the Deploy to AWS section, I provide more of an explanation of what goes into this file.  Mostly it's just the credentials that are passed into the Cloudformation template.

> Did this cover the case when there's more than 50 users and one of the last ones gets deleted?

I verified that the endpoint we use now returns the full set of users in AWS Identity center.  When testing, I used a broad range of people to add and remove from the test group.

Happy to answer any other questions about the deployment in the private channel!

yugoPanorama commented:
> We use the [SAM CLI](https://github.com/aws/aws-sam-cli) to deploy the infrastructure changes here. Deploying with SAM uses a `samconfig.toml` file which we don't commit to source code, since it has sensitive creds. [On panopedia](https://panoramaed.atlassian.net/wiki/spaces/ENG/pages/2833842186/AWS+SSO+Sync#Deploy-to-AWS) in the Deploy to AWS section, I provide more of an explanation of what goes into this file. Mostly it's just the credentials that are passed into the Cloudformation template.

Thanks!  Would it make sense to add `samconfig.toml` to the `.gitignore` file, or would that not be a good fit because of the upstream audience?

> > Did this cover the case when there's more than 50 users and one of the last ones gets deleted?
> 
> I verified that the endpoint we use now returns the full set of users in AWS Identity center. When testing, I used a broad range of people to add and remove from the test group.

👍 

> Happy to answer any other questions about the deployment in the private channel!

Thanks!

eswidler commented:
> Thanks! Would it make sense to add samconfig.toml to the .gitignore file, or would that not be a good fit because of the upstream audience?

It looks like there's already an entry limiting what's in `.aws-sam/` but maybe that's just for the build components.  I might hold off on making that change since I'm not sure what the upstream owner will do here.

yugoPanorama commented:
> > Thanks! Would it make sense to add samconfig.toml to the .gitignore file, or would that not be a good fit because of the upstream audience?
> 
> It looks like there's already an entry limiting what's in `.aws-sam/` but maybe that's just for the build components. I might hold off on making that change since I'm not sure what the upstream owner will do here.

Oh! I didn't realize it was in a subdirectory - thanks for pointing that out!  Then it LGTM!
